### ISSUE TITLE
Regular Data Persistence

### ISSUE THREAD

**Issue by HealMah** [ocoda/event-sourcing#395]
First off, though I am absolutely new to all the architectural patterns this library is addressing:   I SEE DEFINITELY HOW GREAT IS YOUR IMPLEMENTATION!
Kudo to You, You are Great!!

Now, because I am new to all these and given your github example, I need a little help to understand:

1. How to connect this with the regular Mongoose models and schema pattern and save to them.
2. Is the aggregate repository the right place to save to my db models as well?
3. Or, is the persistence event/repository collections provided in the library config enough?

Please, pardon my shortcomings! 
But how to go this is not clear.
Hope you understand and will put me on the right footing. THANKS.

**Comment by drieshooghe** [comment]
Hi @HealMah 

Thanks! Regarding your questions:
1. There isn't a Mongoose version of the library, the `@ocoda/event-sourcing-mongodb` package uses the official MongoDB driver under the hood. All persistence and fetching of events happens under the hood, so it doesn't expose any models.

2. Your aggregate repository should only be responsible for:
- persisting your aggregate (by committing events to the event-store)
- fetching your aggregate (by fetching those committed events and reapplying them on your aggregate class)

  In other words: creating read-models (projections) is out of scope for this library (for now). If you wish to do this, you could create an event-publisher that publishes events committed to the event-store to a service that is responsible for creating projections.
The documentation is unfortunately incomplete, but the general idea can be seen [here](https://ocoda.github.io/event-sourcing/advanced/event_pubsub).

Hoping that explains things a little.

**Comment by SemAiSoft** [comment]
> Hi @HealMah
> 
> Thanks! Regarding your questions:
> 
> 1. There isn't a Mongoose version of the library, the `@ocoda/event-sourcing-mongodb` package uses the official MongoDB driver under the hood. All persistence and fetching of events happens under the hood, so it doesn't expose any models.
> 2. Your aggregate repository should only be responsible for:
> 
> * persisting your aggregate (by committing events to the event-store)
> * fetching your aggregate (by fetching those committed events and reapplying them on your aggregate class)
>   In other words: creating read-models (projections) is out of scope for this library (for now). If you wish to do this, you could create an event-publisher that publishes events committed to the event-store to a service that is responsible for creating projections.
>   The documentation is unfortunately incomplete, but the general idea can be seen [here](https://ocoda.github.io/event-sourcing/advanced/event_pubsub).
> 
> Hoping that explains things a little.

Thanks for the reply. I see a sample git example with the following - though using native `nest-cqrs`. Is there anything wrong with this approach? Is it anti-pattern?

```
@Injectable()
export class OrdersRepository {
  constructor(@InjectModel('Order') private readonly model: Model<OrderDocument>) {}

  async findAll() {
    const docs = await this.model.find().exec();
    const orderRoot = new OrderRoot(undefined);
    orderRoot.setData(docs);
    orderRoot.foundOrder();
    return orderRoot;
  }

  async findOne(id: Types.ObjectId) {
    const doc = await this.model.findById(id);
    const orderRoot = new OrderRoot(id.toHexString());
    if (doc === null) {
      throw new NotFoundException(`Order with id ${id} does not exists`);
    }
    orderRoot.setData(doc);
    orderRoot.foundOrder();
    return orderRoot;
  }

  async createOne(dto: CreateOrderDto) {
    const doc = await this.model.create(dto);
    const orderRoot = new OrderRoot(doc.id);
    orderRoot.setData(doc);
    return orderRoot;
  }

  async updateOne(id: string, dto: UpdateOrderDto) {
    const doc = await this.model.findByIdAndUpdate(new Types.ObjectId(id), { $set: dto }, { new: false }).exec();
    const orderRoot = new OrderRoot(id);
    if (doc === null) {
      throw new NotFoundException(`Order with id ${id} does not exists`);
    }
    orderRoot.setData(doc);
    return orderRoot;
  }

  async deleteOne(id: Types.ObjectId) {
    const doc = this.model.findByIdAndDelete(id).exec();
    const orderRoot = new OrderRoot(id.toHexString());

    if (doc === null) {
      throw new NotFoundException(`Order with id ${id} does not exists`);
    }

    orderRoot.setData(id.toHexString());
    return orderRoot;
  }
}
```

**Comment by drieshooghe** [comment]
Hi @SemAiSoft

There's nothing wrong with that example if you just want to use CQRS, but it doesn't follow an event-sourcing pattern.
